addons/mail/models/discuss/discuss_channel.py

```diff
@@ -161,6 +166,19 @@ def _constraint_group_id_channel(self):
 
     # COMPUTE / INVERSE
 
+    @api.depends("channel_member_ids", "auto_recompute_name")
+    def _compute_name(self):
+        for channel in self.filtered(lambda c: c.auto_recompute_name):
+            new_name = ", ".join(
+                channel.channel_member_ids.mapped(
+                    lambda m: m.partner_id.name if m.partner_id else m.guest_id.name,
+                )[:5],
```

Comment:
We'd better not fetching all the member ids only to ignore them later on. I'd directly use search + limit.